thread/thread.cpp

```diff
@@ -1319,9 +1325,14 @@ R"(
     {
         if (unlikely(!th))
             LOG_ERROR_RETURN(EINVAL, , "invalid parameter");
-        if (unlikely(th->state != states::SLEEPING)) return;
+        if (unlikely(th->state != states::SLEEPING)) {
+        out: // may have thread_yield()-ed
+            if (th->state != states::READY || th->error_number == 0)
+                th->error_number = error_number;
```

Comment:
multi vcpu condition may breaks interrupt error_number.

due to visibility, and non-atomic check and set for `th->state` and `th->error_number`